3_core_core_analysis/7_stability_vs_genome_location.py

```diff
@@ -0,0 +1,384 @@
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:core_acc] *
+#     language: python
+#     name: conda-env-core_acc-py
+# ---
+
+# # Stability vs genome location
+#
+# Here we want to know about the relationship between stability and genome distance: Are more stable genes located in similar locations on the genome in PAO1 vs PA14 compared to least stable genes?
+#
+# We hypothesize that least stable genes are not syntenic (i.e. located in different regions of the genome) across strains, which might indicate a different some transcriptional re-wiring across strain types.
+#
+# There are 2 approaches we take here:
+# 1. For each least/most stable gene, get the neighboring core/homologous genes and determine if they match between PAO1 vs PA14.
+# 2. Scale PA14 gene ids to same range as PAO1 then take distance between PAO1 gene and homolog of PAO1 gene
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+import os
+import random
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+from scripts import paths, utils
+
+random.seed(1)
+# -
+
+# User params
+# Number of neighboring core genes to look at
+# [- window, + window]
+window = 10
+
+# ## Load data
+
+# Input similarity scores and annotations filenames
+# Since the results are similar we only need to look at the scores for one strain type
+pao1_similarity_filename = "pao1_core_similarity_associations_final_spell.tsv"
+pa14_similarity_filename = "pa14_core_similarity_associations_final_spell.tsv"
+
+# Import df
+pao1_similarity = pd.read_csv(pao1_similarity_filename, sep="\t", index_col=0, header=0)
+pa14_similarity = pd.read_csv(pa14_similarity_filename, sep="\t", index_col=0, header=0)
+
+print(pao1_similarity.shape)
+pao1_similarity.head()
+
+print(pa14_similarity.shape)
+pa14_similarity.head()
+
+# ## Get most and least stable genes
+
+pao1_least_stable = pao1_similarity[pao1_similarity["label"] == "least stable"].index
+pao1_most_stable = pao1_similarity[pao1_similarity["label"] == "most stable"].index
+
+# ## Get mapping from PAO1 to PA14 ids
+
+pao1_annotation_filename = paths.GENE_PAO1_ANNOT
+gene_mapping_pao1 = utils.get_pao1_pa14_gene_map(pao1_annotation_filename, "pao1")
+
+gene_mapping_pao1.head()
+
+# Select only genes with 1-1 mapping
+gene_mapping_pao1 = gene_mapping_pao1.query("num_mapped_genes==1")
+# Select only PA14 column
+gene_mapping_pao1 = gene_mapping_pao1["PA14_ID"].to_frame()
+
+gene_mapping_pao1.head()
+
+
+# ## Approach 1
+#
+# For each most and least stable gene, select the neighboring core genes and determine if the neighboring core genes in PAO1 match with the ones in PA14. If the proprotion of matches is high then that indicates that the most/least stable genes is located in the same relative position on the genome in PAO1 and PA14.
```

Comment:
```suggestion
# For each most and least stable gene, select the neighboring core genes and determine if the neighboring core genes in PAO1 match with the ones in PA14. If the proprotion of matches is high then that indicates that the most/least stable genes are located in the same relative position on the genome in PAO1 and PA14.
```

My change can go either way: e.g., you are talking about one single gene or you are talking about multiple genes. Assuming the latter, but feel free to correct
```suggestion
# For each most and least stable gene, select the neighboring core genes and determine if the neighboring core genes in PAO1 match with the ones in PA14. If the proprotion of matches is high then that indicates that the most/least stable genes is located in the same relative position on the genome in PAO1 and PA14.
```
```suggestion
# For each most and least stable gene, select the neighboring core genes and determine if the neighboring core genes in PAO1 match with the ones in PA14. If the proprotion of matches is high then that indicates that the most/least stable genes is located in the same relative position on the genome in PAO1 and PA14.
```